src/Build/Evaluation/ItemSpec.cs

```diff
@@ -446,6 +461,12 @@ public GlobFragment(string textFragment, string projectDirectory)
         {
         }
 
+        public override IEnumerable<string> GetReferencedItems()
+        {
+            // This fragment cannot efficiently enumerate its referenced items.
+            return null;
```

Comment:
Should this throw NotImplementedException to protect against future use where someone hopes this works?

Reply:
As I understand it, we _do_ use this; we just haven't optimized for it. If it's easy to batch things, we do. If not, this returns null, and we do the slow thing. That shouldn't hurt perf too much because I don't know of cases when someone does a long series of RemoveOperations with wildcards, though I may be wrong on that point.

Reply:
Yes, this is intentional. https://github.com/dotnet/msbuild/pull/6716#discussion_r681603275